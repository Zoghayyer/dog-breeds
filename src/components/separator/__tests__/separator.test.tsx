import * as React from 'react';
import { shallow } from 'enzyme';
import Separator, { SeparatorProps } from '../separator';

describe('(Component) Separator', () => {
  let component;
  const defaultProps: SeparatorProps = {
    type: 'horizontal',
  };

  const shallowSeparator = (extendedProps: Partial<SeparatorProps> = {}) => {
    component = shallow(<Separator {...defaultProps} {...extendedProps} />);
  };

  beforeEach(() => shallowSeparator());

  it('renders without crashing', () => {
    shallow(<Separator {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });

  it('renders the separator component when type is horizontal', () => {
    shallowSeparator({ type: 'horizontal'});
    expect(component).toMatchSnapshot();
  });

  it('renders the separator component when type is vertical', () => {
    shallowSeparator({ type: 'vertical'});
    expect(component).toMatchSnapshot();
  });
});
