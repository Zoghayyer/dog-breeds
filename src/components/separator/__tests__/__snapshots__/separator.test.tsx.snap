// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Separator renders the separator component when type is horizontal 1`] = `
<div
  className="horizontal-separator"
/>
`;

exports[`(Component) Separator renders the separator component when type is vertical 1`] = `
<div
  className="vertical-separator"
/>
`;

exports[`(Component) Separator renders without crashing 1`] = `
<div
  className="horizontal-separator"
/>
`;
